File: backend/app/schemas/pending_skill_creation.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class PendingSkillCreationResponse(BaseModel):
    id: uuid.UUID
    session_id: uuid.UUID
    tool_use_id: str | None = None
    detected_name: str
    resolved_name: str | None
    description: str | None
    workspace_files_prefix: str | None
    skill_relative_path: str
    status: str
    skill_id: int | None
    error: str | None
    result: dict | None = None
    created_at: datetime
    updated_at: datetime


class PendingSkillCreationConfirmRequest(BaseModel):
    resolved_name: str | None = None
    description: str | None = Field(default=None, max_length=1000)
    overwrite: bool = False


class PendingSkillCreationCancelRequest(BaseModel):
    reason: str | None = Field(default=None, max_length=1000)

File: frontend/features/chat/types/api/pending-skill-creation.ts
export type PendingSkillCreationStatus =
  | "pending"
  | "creating"
  | "success"
  | "failed"
  | "canceled";

export interface PendingSkillCreation {
  id: string;
  session_id: string;
  tool_use_id?: string | null;
  detected_name: string;
  resolved_name?: string | null;
  description?: string | null;
  workspace_files_prefix?: string | null;
  skill_relative_path: string;
  status: PendingSkillCreationStatus;
  skill_id?: number | null;
  error?: string | null;
  result?: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
}

export interface PendingSkillCreationConfirmInput {
  resolved_name?: string | null;
  description?: string | null;
  overwrite?: boolean;
}

export interface PendingSkillCreationCancelInput {
  reason?: string | null;
}
